app/models/piece.rb

```diff
@@ -15,22 +15,68 @@ def serializable_hash(options = nil)
     super.merge("type" => type)
   end
 
-  def move_to!(row, col)
-    raise "Out of bounds" if row < 0 || row > 7 || col < 0 || col > 7
-    create_move!(row, col)
-    update(row: row, col: col)
-  end
+  def move_to!(to_row, to_col)
+    # rubocop: disable LineLength
+    Rails.logger.debug "Model move_to. from: #{row},#{col} to: #{to_row},#{to_col}"
+
+    if valid_move?(to_row, to_col)
```

Comment:
That makes a ton of sense, except the controller update method simply calls piece.move_to! and doesn't check a return type.

We probably need to have move_to! return a boolean though.

A true return causes the player turn to toggle and perhaps a notice 'white player turn' || 'black player turn'.

A false return causes the piece to snap back to its 'from' location and perhaps a notice that the attempted move is invalid - and a reason.